graph/number_of_distinct_islands.cpp

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include <vector>
+#include <set>
+#define int int64_t
+using namespace std;
+
+void dfs(vector<vector<int>> &grid, vector<vector<int>> &visited, int i, int j, int base_i, int base_j, int m, int n, vector<pair<int, int>> &temp)
```

Comment:
Functions should be documented explaining the use of its parameters, a brief description, and what it returns.